### PR TITLE
feat(integrations): add support for SAP successfactors

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe the problem and your solution 

- Add support for SAP successfactors
- The docs is a little bit long as we need to generate an SAML assertion, which is a more involving step.
- Also added a new `grant_type: urn:ietf:params:oauth:grant-type:saml2-bearer` so that validation would pass.

### CODE REVIEW

File: docs-v2/integrations/all/sap-success-factors.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+| Per-customer configurations | ✅ |
+</Accordion>
+</AccordionGroup>
+
+<PreBuiltUseCases />
+
+## Access requirements
+| Pre-Requisites | Status | Comment|
+| - | - | - |
+| Paid dev account | ❓ |  |
```
Comment (khaliqgant) [inline docs-v2/integrations/all/sap-success-factors.mdx]: Can you fill out this information?

File: docs-v2/integrations/all/sap-success-factors.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+| Paid test account | ❓ |  |
+| Partnership | ❓ | |
+| App review | ❓ |  |
+| Security audit | ❓ | |
+
+
+## Setup guide
+
+_No setup guide yet._
```
Comment (khaliqgant) [inline docs-v2/integrations/all/sap-success-factors.mdx:27]: Can you provide this information? You can see hubspot as an example https://docs.nango.dev/integrations/all/hubspot#setup-guide

File: docs-v2/integrations/all/sap-success-factors.mdx (hunk 1 above)
Comment (hassan254-prog) [reply docs-v2/integrations/all/sap-success-factors.mdx]: Updated in the latest commit.

File: docs-v2/integrations/all/sap-success-factors.mdx (hunk 2 above)
Comment (hassan254-prog) [reply docs-v2/integrations/all/sap-success-factors.mdx:27]: I have added a link to the connect-ui documentation, as it contains detailed information on how to achieve this.

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**